.projenrc.js

```diff
@@ -105,3 +115,74 @@ project.eslint.addRules({
 });
 
 project.synth();
+
+function codeGenFetchAssemblies() {
+  const scriptPath = "scripts/fetch-assemblies.js";
+
+  const packagesPath = "public/packages";
+
+  const script = new SourceCode(project, scriptPath);
+  script.line(`// ${FileBase.PROJEN_MARKER}`);
+  script.line(fromEnv.toString());
+  script.line(fetchAssembly.toString());
+  script.line(
+    `${fetchAssembly.name}(${fromEnv.name}('PACKAGE_NAME'), ${fromEnv.name}('PACKAGE_VERSION'), '${packagesPath}')`
+  );
+
+  project.gitignore.exclude(packagesPath);
+
+  const task = project.addTask("dev:fetch-assemblies");
+  task.exec(`node ${scriptPath}`);
+}
+
+function fromEnv(name) {
+  const value = process.env[name];
+  if (!value) {
+    throw new Error(`Missing ${name} env variable`);
+  }
+  return value;
+}
+
+/**
+ * Fetch an assembly by name and version, accepted via env variable for easier invocation.
+ * This fetches all the necessary assemblies into the packages path, which can later be used to instantiate
+ * a jsii type system for the specific package.
+ */
+function fetchAssembly(packageName, packageVersion, packagesPath) {
+  const exec = require("child_process").exec;
+  const fs = require("fs");
+  const os = require("os");
+  const path = require("path");
+
+  if (packageVersion.startsWith("^")) {
+    packageVersion = packageVersion.substring(1, packageVersion.length);
+  }
+
+  const output = `${packagesPath}/${packageName}@${packageVersion}/jsii.json`;
+
+  const tempDir = fs.mkdtempSync(`${os.tmpdir()}/assembly`);
```

Comment:
Should this temporary directory be cleaned up after the assemblies are copied successfully?

Reply:
I'm usually not a stickler for it since temp is purged on restart...but ok :)